tsl/src/dist_util.c

```diff
@@ -113,6 +127,27 @@ dist_util_set_id(Datum dist_id)
 					 (errmsg("database is already a member of a distributed database"))));
 	}
 
+	if (check_metadata)
+	{
+		uuid = ts_metadata_get_value(CStringGetDatum(METADATA_UUID_KEY_NAME),
+									 CSTRINGOID,
+									 UUIDOID,
+									 &isnull);
+		if (isnull)
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_DATA_NODE_INVALID_CONFIG),
+					 (errmsg("invalid configuration state for data node"),
+					  errdetail("The data node is not part of a distributed database."))));
+		else if (uuid_matches(uuid, dist_id))
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_DATA_NODE_INVALID_CONFIG),
+					 (errmsg("cannot add the current database as a data node"),
+					  errdetail("A database that serves as an access node cannot simultaneously be "
+								"a data node."),
```

Comment:
I'm not sure this error message is capturing the issue here.  If a database is set up as an access node, it will fail the `dist_util_membership() != DIST_MEMBER_NONE` check above, this check here specifically catches the case when a new distributed database is being created by adding a nondistibuted database as a data node to itself.

Reply:
Yes. That is what my first version of the message said. Reverted to something along those lines again and provided more elaborate details and hints.